pkgs/top-level/lua-packages.nix

```diff
@@ -64,7 +64,8 @@ in
   };
 
   luarocks = callPackage ../development/tools/misc/luarocks/default.nix {
-    inherit lua lib;
+    inherit lib;
+    inherit (pkgs.buildPackages) lua;
```

Comment:
This will be a wrong lua when using g a different lua version

```
  ### LUA interpreters
  luaInterpreters = callPackage ./../development/interpreters/lua-5 {};
  inherit (luaInterpreters) lua5_1 lua5_2 lua5_2_compat lua5_3 lua5_3_compat lua5_4 lua5_4_compat luajit_2_1 luajit_2_0;

  lua5 = lua5_2_compat;
  lua = lua5;

  lua51Packages = recurseIntoAttrs lua5_1.pkgs;
  lua52Packages = recurseIntoAttrs lua5_2.pkgs;
  lua53Packages = recurseIntoAttrs lua5_3.pkgs;
  luajitPackages = recurseIntoAttrs luajit.pkgs;

  luaPackages = lua52Packages;

  luajit = luajit_2_1;

  luarocks = luaPackages.luarocks;
  luarocks-nix = luaPackages.luarocks-nix;

  toluapp = callPackage ../development/tools/toluapp {
    lua = lua5_1; # doesn't work with any other :(
  };

  ### END OF LUA
```

Fix would be using makeScopeWithSplicing

Reply:
Right, but I don't know how to do this properly.

Reply:
i'll try

Reply:
https://github.com/NixOS/nixpkgs/pull/187934

Reply:
Ok, I dropped this commit.